pkg/platform/api/graphql/model/buildplanner/buildExpression.go

```diff
@@ -74,23 +111,36 @@ func NewBuildExpression(data []byte) (*BuildExpression, error) {
 		return nil, errs.Wrap(err, "Could not get solve node")
 	}
 
-	requirementsNode, err := getRequirementsNode(expression)
+	requirementsNode, err := getRequirementsNode(solveNode)
```

Comment:
Note this is the only place where `getRequirementsNode` is called, and it always does the JSON marshalling right after. I would argue the json marshalling should just be part of `getRequirementsNode()`.